test/suites/machines/machines_test.go

```diff
@@ -0,0 +1,251 @@
+						"app": "my-app",
+					},
+					Annotations: map[string]string{
+						karpv1.DoNotDisruptAnnotationKey: "true",
+					},
+				},
+				// Each node has 8 cpus, so should fit 2 pods.
+				ResourceRequirements: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceCPU: resource.MustParse("3"),
+					},
+				},
+				TerminationGracePeriodSeconds: lo.ToPtr[int64](0),
+			},
+		})
+		selector = labels.SelectorFromSet(dep.Spec.Selector.MatchLabels)
+	})
+
+	It("should have networking labels applied by machine api", func() {
+		// Check if networking settings are compatible with this test
+		// NETWORK_PLUGIN_MODE must be overlay, but NETWORK_PLUGIN and NETWORK_DATAPLANE
+		// can be unset (using defaults) or set to expected values (but not set to different values)
+		settings := env.ExpectSettings()
+
+		// Helper function to check if env var is unset or set to expected value
+		checkEnvVar := func(envName, expectedValue string) bool {
+			for _, env := range settings {
+				if env.Name == envName {
+					return env.Value == expectedValue
+				}
+			}
+			return true // Not set is acceptable
+		}
+
+		usingCompatiblePlugin := checkEnvVar("NETWORK_PLUGIN", consts.NetworkPluginAzure)
+		usingExpectedPluginMode := lo.Contains(settings, corev1.EnvVar{Name: "NETWORK_PLUGIN_MODE", Value: consts.NetworkPluginModeOverlay})
+		usingCompatibleDataplane := checkEnvVar("NETWORK_DATAPLANE", consts.NetworkDataplaneCilium)
+
+		if !(usingCompatiblePlugin && usingExpectedPluginMode && usingCompatibleDataplane) {
+			Skip("TODO: generalize test for any networking configuration. Skipping as not in expected config for the test")
+		}
+
+		env.ExpectCreated(nodeClass, nodePool, dep)
+
+		node := env.EventuallyExpectCreatedNodeCount("==", 1)[0]
+		env.EventuallyExpectRegisteredNodeClaimCount("==", 1)
+		env.EventuallyExpectCreatedMachineCount("==", 1)
+		env.EventuallyExpectHealthyPodCount(selector, int(numPods))
+
+		// Check that the node has the expected networking labels
+		Expect(node.Labels).To(HaveKeyWithValue("kubernetes.azure.com/ebpf-dataplane", consts.NetworkDataplaneCilium))
+		Expect(node.Labels).To(HaveKeyWithValue("kubernetes.azure.com/azure-cni-overlay", "true"))
+		Expect(node.Labels).To(HaveKeyWithValue("kubernetes.azure.com/podnetwork-type", consts.NetworkPluginModeOverlay))
+
+		// Note: these labels we only check their keys since, the values are dynamic
+		// TODO: improve E2E test to be dynamic, reusing the same provisioning logic we have for labels creation.
+		Expect(lo.Keys(node.Labels)).To(ContainElements([]string{
+			"kubernetes.azure.com/network-subnet",
+			"kubernetes.azure.com/nodenetwork-vnetguid",
+			"kubernetes.azure.com/network-stateless-cni",
+		}))
+	})
+
+	// NOTE: ClusterTests modify the actual cluster itself, which means that preforming tests after a cluster test
+	// might not have a clean environment, and might produce unexpected results. Ordering of cluster tests is important
```

Comment:
minor: Should we add a note here that this is safe in CI because each E2E runs on its own cluster?

Reply:
So, its safe so long as the suite itself is setup correctly to handle the ordering. However, there is leakage between test cases within the suite